### PR TITLE
Feat [CHK-3322] Added random query param to url when creating new wallet

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
Note for reviewers
Sonarcloud error not added in this PR

#### List of Changes
Added a random query param to the returned url

#### Motivation and Context
We need to use a random query param to avoid caching

#### How Has This Been Tested?
Unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ class WalletController(
                     .redirectUrl(
                         UriComponentsBuilder.fromUri(returnUri)
                             .fragment(
-                                "walletId=${walletId}&useDiagnosticTracing=${request.useDiagnosticTracing}&paymentMethodId=${request.paymentMethodId}"
+                                "walletId=${walletId}&useDiagnosticTracing=${request.useDiagnosticTracing}&paymentMethodId=${request.paymentMethodId}&v=${Random.nextInt(1, 100000000)}"
```
[the remark this answers is not in the document]
Comment (scaminati-bv) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:74]: You can use this

`.queryParam("t", Instant.now().toEpochMilli())`

[the remark this answers is not in the document]
Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:74]: this was a query param here 5ee146a8afb500d4dd561202d8906a30ab80ca96
but i was told to edit the fragment instead

[the remark this answers is not in the document]
Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:74]: I will use your random generator tho

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 2]
```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
...
+            .value<String> { walletId ->
+                // Assert that the walletId is as expected
+                assert(walletId.startsWith(WALLET_DOMAIN.id.value.toString().trim())) {
+                    "walletId is not the expected value"
+                }
+            }
+            .jsonPath("$.redirectUrl")
+            .value<String> { redirectUrl ->
+                // Assert that the redirectUrl starts with the expected base URL
+                assert(redirectUrl.contains("#walletId=${WALLET_DOMAIN.id.value}")) {
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:132]:
```suggestion
                assertTrue(redirectUrl.contains("#walletId=${WALLET_DOMAIN.id.value}")) {
```

use junit jupiter assertTrue message here to have test failure report being populated correctly

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 3]
```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
...
+                    "Redirect URL does not contains the expected walletId in fragment"
+                }
+                // Check for the presence of other parameters
+                assert(
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:136]:
```suggestion
                assertTrue(
```
same as above

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 4]
```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
...
+                    redirectUrl.contains(
+                        "useDiagnosticTracing=${WalletTestUtils.CREATE_WALLET_REQUEST.useDiagnosticTracing}"
+                    )
+                ) {
+                    "Redirect URL does not contain the expected useDiagnosticTracing parameter"
+                }
+                assert(
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:143]:
```suggestion
                assertTrue(
```
same as above

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 5]
```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
...
+            .jsonPath("$.walletId")
+            .value<String> { walletId ->
+                // Assert that the walletId is as expected
+                assert(walletId.startsWith(WALLET_DOMAIN.id.value.toString().trim())) {
+                    "walletId is not the expected value"
+                }
+            }
+            .jsonPath("$.redirectUrl")
+            .value<String> { redirectUrl ->
+                // Assert that the redirectUrl starts with the expected base URL
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:131]: Misleading comment, we are not checking the url base path in the received response here.
Add missing check here

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt [hunk 6]
```diff
@@ -119,15 +119,35 @@ class WalletControllerTest {
...
+                    "Redirect URL does not contain the expected useDiagnosticTracing parameter"
+                }
+                assert(
+                    redirectUrl.contains(
+                        "paymentMethodId=${WalletTestUtils.CREATE_WALLET_REQUEST.paymentMethodId}"
+                    )
+                ) {
+                    "Redirect URL does not contain the expected paymentMethodId parameter"
+                }
+            }
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt:150]: we are not checking that returned url contains the newly added query parameter in this test. Please add missing verification here

**Review by pietro-tota — APPROVED** [review]
Looks good. Just some trivial issue about tests

File: src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt (hunk 2 above)
Comment (EmanueleBVtech) [reply src/test/kotlin/it/pagopa/wallet/controllers/WalletControllerTest.kt]: I respond only on this comment, 
i'm accepting all of them.

**Review by CianoDanilo — APPROVED** [review]

File: src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt [hunk 7]
```diff
@@ -43,13 +44,13 @@ class WalletController(
     private val webClient: WebClient = WebClient.create(),
 ) : WalletsApi {
 
+    @SuppressWarnings("kotlin:S6508")
```
Comment (CianoDanilo) [inline src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:47]: Please remove this suppressWarnings

Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:47]: done here c862cd6bcc1b4ddb1bb866ee039c4a18bb557003

Comment (CianoDanilo) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt:47]: I think u removed it from the wrong method

Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/wallet/controllers/WalletController.kt]: done here 4288cda0b62d2bebd26b7f3dc045b6653acd84e2

**Review by CianoDanilo — APPROVED** [review]

**Pull request merged**